### PR TITLE
feature: io package inspect stream reader

### PR DESCRIPTION
**Pull request by szuecs**
feature: io package inspect stream reader
refactor: block* filters to use the new io package

last step to replace https://github.com/zalando/skipper/pull/2428

### CODE REVIEW

File: filters/block/block_test.go [hunk 1]
```diff
@@ -2,16 +2,16 @@ package block
 
 import (
 	"bytes"
-	"io"
+	stdlibio "io"
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:5]: :cry:

File: filters/block/block_test.go [hunk 2]
```diff
@@ -39,22 +39,26 @@ func TestMatcher(t *testing.T) {
...
 			content: "",
+			block:   []byte(".class"),
 			err:     nil,
 		},
 		{
 			name:    "small string",
 			content: ".class",
-			err:     proxy.ErrBlocked,
+			block:   []byte(".class"),
+			err:     io.ErrBlocked,
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:49]: This is really confusing

File: filters/block/block_test.go [hunk 3]
```diff
@@ -234,9 +246,87 @@ func TestBlock(t *testing.T) {
...
+			t.Errorf("Blocked response status code %d", rsp.StatusCode)
+		}
+	})
+	t.Run("pass request on empty body with filter chain", func(t *testing.T) {
+		args := []interface{}{"foo"}
+		args2 := []interface{}{"bar"}
+		r := &eskip.Route{Filters: []*eskip.Filter{
+			{Name: spec.Name(), Args: args},
+			{Name: spec.Name(), Args: args2},
+		}, Backend: backend.URL}
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:324]: Why not (here and elsewhere)
```go
r = eskip.MustParse(fmt.Sprintf(`block("foo") -> block("bar") -> "%s"`, backend.URL))
proxy := proxytest.New(fr, r...)
```

File: filters/block/block_test.go [hunk 4]
```diff
@@ -234,9 +246,87 @@ func TestBlock(t *testing.T) {
...
+		proxy := proxytest.New(fr, r)
+		defer proxy.Close()
+
+		buf := bytes.NewBufferString("")
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:328]: nil should also work

File: filters/block/block_test.go [hunk 5]
```diff
@@ -234,9 +246,87 @@ func TestBlock(t *testing.T) {
...
+		defer proxy.Close()
+
 		buf := bytes.NewBufferString("")
-		req, err := http.NewRequest("POST", reqURL.String(), buf)
+		req, err := http.NewRequest("POST", proxy.URL, buf)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		rsp, err := http.DefaultClient.Do(req)
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:309]: Should use proxy.Client()

File: filters/block/block_test.go (hunk 2 above)
Comment (szuecs) [reply filters/block/block_test.go:49]: What do you mean?

File: filters/block/block_test.go (hunk 4 above)
Comment (szuecs) [reply filters/block/block_test.go:328]: does it matter?

File: filters/block/block_test.go (hunk 1 above)
Comment (szuecs) [reply filters/block/block_test.go:5]: cmon

File: filters/block/block_test.go (hunk 3 above)
Comment (szuecs) [reply filters/block/block_test.go:324]: It was to easily copy and paste and maybe also do a table test, but I will change as I did not do the table for different block* filters

File: filters/block/block_test.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply filters/block/block_test.go:49]: I mean there are io Err* variable in stdlib https://pkg.go.dev/io#pkg-variables and this import is confusing

File: filters/block/block_test.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply filters/block/block_test.go:5]: Lets at least use `skpio "github.com/zalando/skipper/io"` import

File: filters/block/block_test.go (hunk 3 above)
Comment (szuecs) [reply filters/block/block_test.go:324]: blockHexContent tests can not be done like this, that's why I did both test cases similar.

File: filters/block/block_test.go (hunk 5 above)
Comment (szuecs) [reply filters/block/block_test.go:309]: done

File: filters/block/block_test.go (hunk 4 above)
Comment (AlexanderYastrebov) [reply filters/block/block_test.go:304]: just shorter

File: filters/block/block_test.go (hunk 2 above)
Comment (szuecs) [reply filters/block/block_test.go:50]: ok, fixed

File: filters/block/block_test.go (hunk 1 above)
Comment (szuecs) [reply filters/block/block_test.go]: done

File: filters/block/block_test.go (hunk 4 above)
Comment (szuecs) [reply filters/block/block_test.go:304]: The pattern would differ to other tests, but this you seem to like :D
Fixed

File: filters/block/block.go [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
+	"bytes"
 	"encoding/hex"
-	"errors"
 
 	"github.com/zalando/skipper/filters"
-)
-
-var (
-	ErrClosed = errors.New("reader closed")
+	"github.com/zalando/skipper/io"
```
Comment (AlexanderYastrebov) [inline filters/block/block.go:8]: I'd still use skpio here

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

File: filters/block/block.go (hunk 6 above)
Comment (szuecs) [reply filters/block/block.go:8]: https://github.com/zalando/skipper/pull/2854

**Pull request merged**